File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/06.Nested-Loop/03-additional/03.luckyNumbers.js
function luckyNumbers(input) {

    let num = Number(input[0]);
    let printline = '';

    for (let i = 1; i <= 9; i++) {
        for (let j = 1; j <= 9; j++) {
            for (let k = 1; k <= 9; k++) {
                for (let l = 1; l <= 9; l++) {
                    if ((i + j == k + l) && num % (i + j) == 0) {
                        printline += `${i}${j}${k}${l} `;
                    }
                }
            }
        }
    }

    console.log(printline);
}
luckyNumbers(['7']);

File: Python_Basics/Homeworks-And-Labs/06.nested_loops/03.additional/03.lucky_numbers.py
num = int(input())
print_line = ''

for i in range(1, 10):
    for j in range(1, 10):
        for k in range(1, 10):
            for l in range(1, 10):
                if (i + j == k + l) and num % (i + j) == 0:
                    print_line += f'{i}{j}{k}{l} '

print(print_line)